packages/orchestration/src/exos/chain-hub.js

```diff
@@ -70,15 +70,24 @@ export const CONNECTIONS_KEY = HubName.ChainConnection;
 export const ASSETS_KEY = HubName.ChainAssets;
 
 /**
- * Character used in a connection tuple key to separate the two chain ids. Valid
- * because a chainId can contain only alphanumerics and dash.
- *
- * Vstorage keys can be only alphanumerics, dash or underscore. That leaves
- * underscore as the only valid separator.
+ * Character used in a connection tuple key to separate the two chain ids.
+ */
+const CHAIN_ID_SEPARATOR = '_';
+
+/**
+ * Vstorage keys can be only alphanumerics, dash, or underscore, which are all
+ * valid characters in chain IDs. So, double each occurence of
+ * {@link CHAIN_ID_SEPARATOR} in the chain ID to make it clear that it's part of
+ * the chain ID rather than a separator.
  *
+ * @param {string} chainId
  * @see {@link https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-2.md}
  */
-const CHAIN_ID_SEPARATOR = '_';
+export const sanitizeChainId = chainId =>
+  chainId.replaceAll(
+    CHAIN_ID_SEPARATOR,
+    `${CHAIN_ID_SEPARATOR}${CHAIN_ID_SEPARATOR}`,
+  );
```

Comment:
on the fence about this suggestion, but since this expression is so close to the const I think it could be clearer:
```suggestion
  chainId.replaceAll('_', '__');
```

Reply:
I see what you mean but I prefer keeping the variable here